app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,20 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user && user.authenticate(params[:session][:password])
```

Comment:
`user && user.authenticate(params[:session][:password])` -> `user&.authenticate(params[:session][:password])`
Tìm hiểu `&` khi call method rồi trả lời vô đây e nhé

Reply:
Question 2: hàm authenticate cơ chế hoạt động thế nào

Reply:
**Question1: & khi call method** 
- Khi gọi một phương thức trên một đối tượng, nếu đối tượng đó là nil, Ruby sẽ gây ra lỗi NoMethodError.
- Toán tử &. (safe navigation operator) giúp  tránh lỗi này bằng cách kiểm tra nếu đối tượng là nil trước khi gọi phương thức.
- Nếu đối tượng không phải là nil, phương thức sẽ được gọi bình thường. Nếu đối tượng là nil, toàn bộ biểu thức sẽ trả về nil mà không gây lỗi.
- Giúp mã gọn hơn và dễ đọc hơn so với cách kiểm tra thủ công bằng &&

**Question 2: Cơ chế hoạt động hàm authenticate**
- Khi  gọi phương thức authenticate và truyền vào mật khẩu người dùng nhập vào, phương thức này sẽ so sánh mật khẩu đã nhập với mật khẩu đã mã hóa lưu trong cơ sở dữ liệu.
- Nếu mật khẩu khớp, phương thức authenticate sẽ trả về đối tượng người dùng (user), ngược lại, nó sẽ trả về false.

Reply:
Chưa thấy fix code e?

Reply:
Dạ, em đã update